### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by Lul-Abdifan**
**In this local storage of portofolio:-**
-Store data in local storage
-Load data to form inputs
-Fix eslint error

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -324,38 +321,33 @@ <h2 class="abtMe_Header">About me </h2>
...
+                               <input type="email" id="email"placeholder="Email address" required>
+                            
+                                 <img src="./assets/images/underInput.png"  id="under_input">
+                                 <div id="showError">
+     
+                                 </div>
+                                                           <label for="comment" name="comment"></label> 
+     
+      <textarea rows="7" cols="50" id="comment" maxlength="500" name="comment" placeholder="Enter text here" required>
+     </textarea>
```
Comment (Qoosim) [inline index.html:189]: - [ ] For some reason, I am able to data to local storage without filling out the comment which must also pass validation. Kindly look into it and fix.

validation-error

**Review by Qoosim — CHANGES_REQUESTED** [review]
#    *Status: Required Changes* :recycle: 

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

-  No linter errors. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

- [ ] After filling up all the form inputs, I should be able to send data to the formspree on clicking the submit button and be redirected to the formspree welcome page. But after several clicking, I was unable to redirect to the formspree page. Kindly look into this and fix it.
your-form

File: index.html (hunk 1 above)
Comment (Lul-Abdifan) [reply index.html:189]: Thank you bro for your feedback.It is not passing empty comment on my machine.Do you mind if you guide me to what you want ?

**Comment by Lul-Abdifan** [comment]
@Qoosim Thank you  .:+1:I appreviate your careful feedback.:smile:Happy Coding.

**Comment by Lul-Abdifan** [comment]
**Required Changes Done** :smile:
-Update to send data to fromspree from
-Update not to store to local storage for unfilled input
-Fix eslint ,conflict,stylelint errors

File: script.js [hunk 2]
```diff
@@ -231,3 +231,30 @@ formCheck.onsubmit = (e) => {
...
+document.getElementById('formSubmit').addEventListener('submit', () => {
+  const datas = {
+    name: nameInput.value.trim(),
+    email: emailInput.value.trim(),
+    comment: commentInput.value.trim(),
+  };
+  if (datas.name && datas.email && datas.comment) {
+    localStorage.setItem('storedData', JSON.stringify(datas));
+  }
+});
```
Comment (christianbiring1) [inline script.js]: - [ ] The user inputs are not supposed to be saved in the local storage only after the user has submitted the form. The save must be async. meaning as the user is filling(typing) at the same time the inputs are being saved in the local storage.
Hint: `You may consider removing this block of code in the submit event` :smile: 

Screenshot from 2022-10-22 13-12-15

File: script.js [hunk 3]
```diff
@@ -231,3 +231,30 @@ formCheck.onsubmit = (e) => {
...
+
+function getStoredData() {
+  const storedData = JSON.parse(localStorage.getItem('storedData'));
+  nameInput.value = storedData.name;
+  commentInput.value = storedData.comment;
+  emailInput.value = storedData.email;
+}
+
+window.addEventListener('load', () => {
```
Comment (christianbiring1) [inline script.js]: - [ ] Kindly make sure you fix this error :smile: 
Hint: `After you remove those lines of code in the submit event everything will be fixed` :smile: 
Screenshot from 2022-10-22 13-11-16

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @Lul-Abdifan , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
# STATUS: REQUEST CHANGES :recycle: 
163947610-e4d849d5-e69c-49b9-90bd-a4191f4e2856

 ## Highlights
- [ ]  All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary`
- [ ] Good commit message :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
**Required changes Done**
-Change submit to change functionality for all inputs
-Fix errors

**Review by AmaduKamara — APPROVED** [review]
# Status: APPROVED 🟢🟢🟢 

Hi @Lul-Abdifan,

Good job on making the changes asked by the previous reviewer 👍 

Your project is complete!  

## Highlights:
✅ You used Gutflow correctly
✅ Linter checks passed
✅ Good commit messages 👍
✅ Form data is stored in local storage as expected 👍 

There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Lul-Abdifan** [comment]
@AmaduKamara  Thank you .:smile:Happy coding:smile:

**Comment by Lul-Abdifan** [comment]
@christianbiring1  Thank you for your feedback.I did well.:smile:happy coding:smile:

**Comment by christianbiring1** [comment]
> @christianbiring1 Thank you for your feedback.I did well.smilehappy codingsmile

You're welcome @Lul-Abdifan :pray: :muscle:

**Pull request merged**